File: App.Models/CaseFileDocument.cs
using App.Core.Models;

namespace App.Models
{
    public class CaseFileDocument : IAuditableEntity
    {
        public int Id { get; set; }
        public int CaseFileId { get; set; }
        public string DocumentPath { get; set; } = string.Empty;
        public string DocumentType { get; set; } = string.Empty;
        public DateTime UploadedAt { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual CaseFile CaseFile { get; set; } = null!;
    }
}

File: App.client/projects/proctor-app/scripts/models/case-file-document.ts
export class CaseFileDocument {
  id: number;
  caseFileId: number;
  documentPath: string;
  documentType: string;
  uploadedAt: string;
  createdOn: string;
  modifiedOn: string;

  constructor() {
    this.id = 0;
    this.caseFileId = 0;
    this.documentPath = this.documentType = '';
    this.uploadedAt = new Date().toISOString();
    this.createdOn = this.modifiedOn = '';
  }
}

